store/principal.go

```diff
@@ -86,13 +86,18 @@ type UpdateUserMessage struct {
 
 // UserMessage is the message for an user.
 type UserMessage struct {
-	ID            int
-	Email         string
-	Name          string
-	Type          api.PrincipalType
-	PasswordHash  string
-	Role          api.Role
-	MemberDeleted bool
+	ID                 int
+	Email              string
+	Name               string
+	Type               api.PrincipalType
+	PasswordHash       string
+	IdentityProviderID *int
+	// IdentityProviderUserInfo is a raw json string from userinfo api response.
+	// e.g. for GitHub's authenticated user api, it looks like the following structure..
+	// https://docs.github.com/en/rest/users/users?apiVersion=2022-11-28#get-the-authenticated-user
+	IdentityProviderUserInfo string
```

Comment:
It's okay if it's string for now, but we need to use a structured proto message for this in the future.